Fix setGasPrice dispatch function parameters

rekmarks opened the pull request:
This fixes a bug introduced in #9599. In that PR, the `gas-price-button-group` prop `handleGasPriceSelection` was overwritten in the swaps gas customization modal in order to take the gas estimate type (a string constant) as its second parameter (context: https://github.com/MetaMask/metamask-extension/pull/9599#discussion_r510516228). At the time, we missed that `handleGasPriceSelection` normally takes as its second parameter a hexadecimal string gas limit. BigNumber errors result.

This PR enables `handleGasPriceSelection` to be used in both contexts by means of named parameters. Note that most of the diff is changing `assert.equal` and `assert.deepEqual` to their `strict` equivalents in unit tests.

rekmarks committed:
Fix setGasPrice dispatch function params

rekmarks committed:
Fix unit tests

danjm commented:
Does `updateCustomGasPrice` on line 213 of `ui/app/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.container.js` also need its parameters updated?

As I read it:
`const updateCustomGasPrice = (newPrice) => dispatch(setCustomGasPrice(addHexPrefix(newPrice)))`
:arrow_down: 
```
 gasPriceButtonGroupProps: {
      ...gasPriceButtonGroupProps,
      handleGasPriceSelection: otherDispatchProps.updateCustomGasPrice,
    },
```
:arrow_down: 
```
<GasPriceButtonGroup
            className="gas-price-button-group--alt"
            showCheck
            {...gasPriceButtonGroupProps}
          />
```
:arrow_down: 
```
  onClick={() =>
          handleGasPriceSelection({
            gasPrice: priceInHexWei,
            gasEstimateType: renderableGasInfo.gasEstimateType,
          })
        }
        key={`gas-price-button-${index}`}
      >
```
So `updateCustomGasPrice` now gets passed on object

danjm reviewed (changes requested):
one change needed I think

rekmarks committed:
Fix gas-modal-page-container gas price selection call

rekmarks commented:
@danjm feedback addressed in 406f06d. The location you highlighted overwrites the `handleGasPriceSelection` prop with a new function, so I just fixed its signature directly. Also updated related tests, with the same `equal` -> `strictEqual` fixes.

rekmarks committed:
Fix mock signature in test

danjm reviewed (approved):
LGTM
